flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/BackPressureStatsTracker.java

```diff
@@ -177,7 +177,7 @@ public boolean triggerStackTraceSample(ExecutionJobVertex vertex) {
 						LOG.debug("Triggering stack trace sample for tasks: " + Arrays.toString(vertex.getTaskVertices()));
 					}
 
-					Future<StackTraceSample> sample = coordinator.triggerStackTraceSample(
+					CompletableFuture<StackTraceSample> sample = coordinator.triggerStackTraceSample(
```

Comment:
Why not you not returning a plain `Future` here? Shouldn't we only explicitly return a `CompletableFuture` if we expect the caller to complete it?